docs/eventsub_docs.md

```diff
@@ -0,0 +1,128 @@
+# EventSub Documentation
+
+EventSub implements the Twitch EventSub notifications. It should cover everything found at https://dev.twitch.tv/docs/eventsub
+
+In contrary to webhooks, eventsub subscriptions do not expire based on time. They will expire when the token used is revoked or if the notification response failure rate is to high.
+
+## Get EventSub Subscriptions
+
+This is an example of how to get eventsub subscriptions.
+
+```go
+client, err := helix.NewClient(&helix.Options{
+    ClientID: "your-client-id",
+    AppAccessToken: "your-app-access-token",
+})
+if err != nil {
+    // handle error
+}
+
+resp, err := client.GetEventSubSubscriptions(&helix.EventSubSubscriptionsParams{
+    Status: helix.EventSubStatusEnabled // This is optional.
+})
+if err != nil {
+    // handle error
+}
+
+fmt.Printf("%+v\n", resp)
+```
+
+## Create EventSub Subscription
+
+To create a subscription call CreateEventSubSubscription with a pointer to a subscription. As of writing Version sould always be "1" except for the Channel Raid Event which is still in beta and therefore you need to use Version "beta".
+Within the Transport the only supported Method currently is "webhook". Callback needs to be a https link on port 443. With the secret you can verify if notifications came from twitch. See (#verify-eventSub-notification)
+
+```go
+client, err := helix.NewClient(&helix.Options{
+    ClientID: "your-client-id",
+    AppAccessToken: "your-app-access-token",
+})
+if err != nil {
+    // handle error
+}
+
+resp, err := client.CreateEventSubSubscription(&helix.EventSubSubscription{
+    Type: helix.EventSubTypeChannelFollow,
+    Version: "1",
+    Condition &helix.EventSubCondition{
+        BroadcasterUserID: "1337"
+    }, 
+    Transport &helix.EventSubTransport{
+        Method: "webhook",
+        Callback: "https://example.com/follow",
+        Secret: "s3cre7w0rd",
+    },
+})
+if err != nil {
+    // handle error
+}
+
+fmt.Printf("%+v\n", resp)
+```
+
+## Delete EventSub Subscription
+
+To delete a subscription you need to call DeleteEventSubSubscription with the subscription id as parameter.
+
+```go
+client, err := helix.NewClient(&helix.Options{
+    ClientID: "your-client-id",
+    AppAccessToken: "your-app-access-token",
+})
+if err != nil {
+    // handle error
+}
+
+resp, err := client.DeleteEventSubSubscription("subscription-id")
+if err != nil {
+    // handle error
+}
+
+fmt.Printf("%+v\n", resp)
+```
+
+
+## Example for handling a notification
+
+```go
+
+type eventSubNotification struct {
+    Subscription    helix.EventSubSubscription `json:"subscription"`
+    Challenge       string `json:"challenge"`
+    Event           json.RawMessage `json:"event"`
+}
+
+func eventsubFollow(w http.ResponseWriter, r *http.Request) {
+    body, err := ioutil.ReadAll(r.Body)
+    if err != nil {
+        logrus.Error(err)
```

Comment:
Please removed all `logrus` package calls and replace with something from the standard library like `log.Println`. This package shouldn't introduce other package functionality.

Reply:
oh dang, i missed this as it was only in the docs, will do!